src/sds_data_model/_raster.py

```diff
@@ -0,0 +1,200 @@
+
+def _get_resample_shape(
+    data_array: DataArray,
+) -> NDArray:
+    return (
+        (
+            (
+                array([data_array.rio.transform().a, -data_array.rio.transform().e])
+                / CELL_SIZE
+            )
+            * data_array.shape
+        )
+        .round()
+        .astype("int")
+    )
+
+
+def _resample_cellsize(
+    data_array: DataArray,
+    categorical: bool = False,
+) -> DataArray:
+    if categorical:
+        interpolation = INTER_NEAREST
+    else:
+        interpolation = INTER_LINEAR
+    resampled = resize(
+        src=data_array.data,
+        dsize=_get_resample_shape(data_array)[::-1],
+        interpolation=interpolation,
+    )
+
+    return _create_data_array(
+        data_array=data_array,
+        data=resampled,
+        geotransform=(
+            f"{data_array.rio.transform().c} {CELL_SIZE} 0 "
+            f"{data_array.rio.transform().f} 0 -{CELL_SIZE}"
+        ),
+    )
+
+
+def _check_no_data(
+    data_array: DataArray,
+    nodata: Optional[float] = None,
+) -> None:
+    if nodata is not None:
+        data_array.rio.write_nodata(nodata, inplace=True)
+    elif data_array.rio.nodata is None:
+        raise Exception(
+            "Input dataset does not have a nodata value. One must be provided."
+        )
+
+
+def _get_bng_offset(
+    data_array: DataArray,
+) -> NDArray:
+    return (
+        (
+            array(
+                [BNG_YMAX - data_array.rio.transform().f, data_array.rio.transform().c]
+            )
+            / CELL_SIZE
+        )
+        .round()
+        .astype("int")
+    )
+
+
+def _to_bng_extent(
+    data_array: DataArray,
+    nodata: Optional[float] = None,
+) -> DataArray:
+    _check_no_data(data_array, nodata)
+
+    offset = _get_bng_offset(data_array)
+
+    bng = full(
+        shape=(array([BNG_YMAX, BNG_XMAX]) / CELL_SIZE).astype("int"),
+        fill_value=data_array.rio.nodata,
+        dtype=data_array.dtype,
+    )
+
+    bng[
+        offset[0] : offset[0] + data_array.shape[0],
+        offset[1] : offset[1] + data_array.shape[1],
+    ] = data_array.data
+
+    return _create_data_array(
+        data_array=data_array,
+        data=bng,
+        geotransform=f"{BNG_XMIN} {CELL_SIZE} 0 {BNG_YMAX} 0 -{CELL_SIZE}",
+    )
+
+
+def _select_data_arrays(
+    dataset: Dataset,
+    band: Union[List[int], List[str], None] = None,
+) -> List[DataArray]:
+    try:
+        data_arrays = [
+            data_array.rename(f"{index + 1}").drop("band")
+            for index, data_array in enumerate(dataset.band_data)
+        ]
+    except AttributeError:
+        data_arrays = [dataset[array_name] for array_name in dataset.rio.vars]
+
+    if band:
+        data_arrays = [
+            data_array
+            for data_array in data_arrays
+            if data_array.name in map(str, band)
+        ]
+    return data_arrays
+
+
+def _resample_and_reshape(
+    dataset: Dataset,
+    band: Union[List[int], List[str], None] = None,
```

Comment:
Same as `_select_data_arrays`: I think this should be bands as it's a collection. Also, I think the type should be Optional

```suggestion
    bands: Optional[Union[List[int], List[str]]] = None,
```